static/test/javascripts/spec/common/commercial/adblock-messages.spec.js

```diff
@@ -0,0 +1,101 @@
+define([
+    'helpers/injector',
+    'common/utils/storage'
+], function (
+    Injector,
+    storage
+) {
+    var injector = new Injector();
+
+    describe('Adblock messages', function () {
+        var adblockMessage, config, detect, userFeatures;
+
+        beforeEach(function (done) {
+            injector.require([
+                'common/modules/commercial/adblock-messages',
+                'common/utils/config',
+                'common/utils/detect',
+                'common/modules/commercial/user-features'
+            ], function () {
+                adblockMessage = arguments[0];
+                config = arguments[1];
+                detect = arguments[2];
+                userFeatures = arguments[3];
+                done();
+            });
+        });
+
+        describe('If adblock banners and messages should be shown', function () {
+            it('should not show adblock messages for the first time users', function () {
+                storage.local.set('gu.alreadyVisited', 0);
+                config.switches.adblock = true;
+                detect.adblockInUse = function () {
```

Comment:
You could reduce some boilerplate by making your beforeEach do something like

    detect.adblockInUse = function () {return mockUserHasAblock;}
    detect.getBreakpoint = function () {return mockBreakpoint;}

and then in your tests,

    mockUserHasAdblock = true;
    mockBreakpoint = 'desktop';

It might make the tests a little more concise.